dlpack support

VoVAllen opened the pull request.

jermainewang commented on tensorflow/c/eager/dlpack.cc:
```diff
@@ -0,0 +1,342 @@
+        "TFE_TensorHandleDevicePointer may not be called on a remote tensor "
+        "handle.");
+    return nullptr;
+  }
```
redundant?

jermainewang commented on tensorflow/c/eager/dlpack.cc:
```diff
@@ -0,0 +1,342 @@
+  }
+  tensorflow::Device* device(absl::get<tensorflow::Device*>(handle->device()));
+  if (device != nullptr) {
+    status->status = device->Sync();
```
Should device be synced here? In RFC discussion, synchronization is performed when a TF tensor is converted to a DLPack tensor.

VoVAllen replied:
You are right. It's copyed from another function and I didn't carefully check it.

jermainewang commented on tensorflow/c/eager/dlpack.cc:
```diff
@@ -0,0 +1,342 @@
+  delete owner;
+}
+
+DLDataType getDLDataType(TF_DataType data_type, TF_Status* status) {
```
`GetDLDataType`

jermainewang commented on tensorflow/c/eager/dlpack.cc:
```diff
@@ -0,0 +1,342 @@
+  return tensor;
+};
+
+void deleter(DLManagedTensor* arg) {
```
`DLManagedTensorDeleter`

jermainewang commented on tensorflow/c/eager/dlpack.cc:
```diff
@@ -0,0 +1,342 @@
+  return dtype;
+}
+
+DLContext getDLContext(TFE_TensorHandle* h, TF_Status* status) {
```
`GetDLContext`

jermainewang commented on tensorflow/c/eager/dlpack.cc:
```diff
@@ -0,0 +1,342 @@
+  int device_id = -1;
+  if (parsed_name.has_id) {
+    device_id = parsed_name.id;
+  }  // Question? device_id?=-1
```
throw error?

jermainewang commented on tensorflow/python/tfe_wrapper.cc:
```diff
@@ -1033,6 +1034,50 @@ PYBIND11_MODULE(_pywrap_tfe, m) {
+          absl::string_view(pycapsule.name()));
+    }
+    TFE_TensorHandle* thandle = tensorflow::TFE_HandleFromDLPack(
+        static_cast<void*>(pycapsule), status.get());
```
No need of `static_cast` for `void*`

jermainewang reviewed (commented):
Overall the code is good to me. I found you've already pushed a PR to TF repo. Let's wait for their response.  Several questions:
* According to the RFC, synchronization will be performed after ToDLPack. Is this correctly implemented?
* I didn't see any python codes. Has this been tested?

pallavigopal committed:
Address the coding style issues due to clang-format version mismatch

galarragas committed:
Expose option to limit the number of partitions that will be delegated to NNAPI

PiperOrigin-RevId: 295962456
Change-Id: I43e13700e23b798ce786b7f1034066961c4c3613

Commit:
Use --std=c++11 flag when targeting iOS platform

PiperOrigin-RevId: 295966567
Change-Id: I6cfb19f70228d8fcb42504430eb2e28beec7c2e7

Commit:
Fix invalid reference due to vector resize.

When creating a new replicate op in the variable runtime reformatting pass, new_replicated_inputs holds references to replicated_inputs, but replicated_inputs could invalidate its elements due to resize.

Reserve enough space to avoid this problem.

PiperOrigin-RevId: 295974084
Change-Id: Ic4f36e90ab1807842ea73d9802b7b58b358b1c98

tensorflower-gardener committed:
Add PopulationCount to the XLA Python client.

PiperOrigin-RevId: 295974574
Change-Id: I8cf982db9b213b530354e01ed982cfa128f80ce8

nhasabni committed:
Merge branch 'master' into nhasabni/dnnl1x_matmul

edloper committed:
Automated rollback of commit ed493143b14c31ebf16881a815e8904e6a82ff9a

PiperOrigin-RevId: 295976413
Change-Id: I24d8fa6b6977fee4d2cb963259bb880775436bc6

tfeher committed:
Merge branch 'master' into allow_build_at_runtime

vinceab committed:
TFLite: static library: fix benchmark build issue

Since the commit ee7642b2670e33a45cc3a6f6585cfab7f7d4f8f6, the benchmark
application is no more building due to the fact that some functions have
been moved.
Add profile_summary_formatter.cc in the PROFILE_SUMMARIZER_SRCS.

Signed-off-by: Vincent ABRIOU <vincent.abriou@st.com>

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 295978060
Change-Id: If0efa7ed880f18575aa7788f877bd1c53c419e8d

tensorflower-gardener committed:
Add "Launch Activities" derived line. And add stats for each step , such as how many kernel/memcpy are launched, what's maximum/avg launch time.

PiperOrigin-RevId: 295978584
Change-Id: Ib2c418ecf034283613e960dd26ddf488ed5ba1bb

tensorflower-gardener committed:
Merge pull request #36741 from tigert1998:patch-1

PiperOrigin-RevId: 295985230
Change-Id: I8286be9f1390e5aa60ae31649a3a290dea5914ac

mrry committed:
[SparseToDense] Fix benchmark reporting for the SparseToDense op.

Previously, we were using a concatenated benchmark arg for the size and rank, which appears to have led to name collisions when the benchmark reporter makes the benchmark name human readable. This change switches the benchmark to using an explicit `ArgPair` for those arguments, which reports them separately.

In addition, this change switches the benchmark to use a DT_INT64 indices tensor, which matches the common case for tf.SparseTensor usage in the TensorFlow API, and avoids a cast in the SparseToDenseOp implementation.

PiperOrigin-RevId: 295985981
Change-Id: I94df39e2ef64b387f4c158ef0c8b33153c3cca4c

aaroey committed:
Fix convert_nodes.cc to support TRT7.

PiperOrigin-RevId: 295986213
Change-Id: I6221fca9278c02c2bb657b844980d2b2aef21a44

gunan committed:
Avoid using hardcoded path in benchmark_test

/tmp does not exist on windows.

PiperOrigin-RevId: 295988695
Change-Id: Ie45e6311df617462f4ba10354fabbeaf2eb05127

ebrevdo committed:
[TF wheel] Put tensorflow & third_party headers in the right directory (again).

Prior to this, headers meant to go into include/{third_party,tensorflow} were being
put into a tensorflow-xxx.data/purelib/include directory during bdist build.  Now they're
placed into tensorflow/include/ and the wheel installs them where they're expected.

PiperOrigin-RevId: 295990673
Change-Id: I0606c75780ae6cda93bc009d3aa5bf03e51e2734

buchgr committed:
Support remote repositories in TF_*_CONFIG_REPO environment variables

Currently TF_*_CONFIG_REPO environment variables point to checked in preconfig packages. After migrating to remote config they will point to remote repositories. The "config_repo_label" function ensures both ways continue to work.

PiperOrigin-RevId: 295990961
Change-Id: I7637ff5298893d4ee77354e9b48f87b8c328c301

karimnosseir committed:
Update test to check for Float 32 check for RandomUniform legalize

PiperOrigin-RevId: 295991291
Change-Id: Ieb1dc23915560bb33d7b036428d0d6bbd81c28ac

kenfranko committed:
Remove values property from DistributedValues.

PiperOrigin-RevId: 295994651
Change-Id: Ic0d003c76e711bee12d5de563de902430e837d5e

tensorflower-gardener committed:
Provide an accessor for dynamic dimension inference.

PiperOrigin-RevId: 295996157
Change-Id: I50ea04cd692d1163b2e05d9f8e12dbeffc11fa3d

Commit:
Move the _TFBufferWrapper helper to a common place instead of in the tpu codebase.

PiperOrigin-RevId: 295996954
Change-Id: I2557fddd7cdc858ce661dd5a8c7bcd9996d519c6

tensorflower-gardener committed:
avoid CHECK in op_level_cost_estimator, silently fallback.

PiperOrigin-RevId: 295998356
Change-Id: Iee5c94376db3cd9d0a69a351eca615b73ec68be9

Commit:
Add a functiondef getter to the context

PiperOrigin-RevId: 296002833
Change-Id: I238a2984a9320c084b7157e6eeb30b30aa132036

VoVAllen committed:
add python api and test

nairb774 committed:
Add access to TEST_UNDECLARED_OUTPUTS_DIR which takes into account Bazel's use
of `/` as a path separator on Windows.

Note: This will have no impact until a later change to use `\` as path separator
is checked in.
PiperOrigin-RevId: 296005736
Change-Id: Id7ada3b06e38399fd17df76fb5c8d3b0ea70e0e2

tensorflower-gardener committed:
Merge pull request #36478 from rahul003:aws_sdk

PiperOrigin-RevId: 296004710
Change-Id: I3f2e9fd5107ccff43fb3e770d9bc607c2fa314a7

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 296006530
Change-Id: I97f75bc86ae4e91f21e2b50c1bcba516d009f297

hawkinsp committed:
[XLA:Python] Add an alias CustomCallWithLayout for CustomCall.

Change in preparation for redefining CustomCall to mean a CustomCall without layout, as in the xla_builder.h C++ API.

PiperOrigin-RevId: 296008567
Change-Id: Id1eac792c5b300f67e04e7055826d8d366993c43

silvasean committed:
Set resource subtype correctly on bound inputs.

This gives more information to shape propagation.

To do this, we need to insert a cast back to the raw imported type, but
shape propagation will eliminate that later.

Incidentally, this also standardizes the arg types used to represent bound
inputs between the V1 (SignatureDef) and V2 (ObjectGraph) importers. As
such, I've fixed a TODO in the verifier to actually verify that the args
have the right type.

PiperOrigin-RevId: 296009270
Change-Id: If5aee852ac08249c37ed8565f2e140e7b54c82d7

ezhulenev committed:
[TF:MLIR] Add pass to move transposes to the end of the block

PiperOrigin-RevId: 296010153
Change-Id: I6b424455d590f97b2c822304dd9a689032f1d7f6

tensorflower-gardener committed:
Merge pull request #33485 from frreiss:issue-data-window-doc

PiperOrigin-RevId: 296010839
Change-Id: Ic65bd56db8b25fe02227066c8fb5c0260827650b

roserg committed:
Add check for "correct" add before inputs reordering.
We can not reorder inputs in broadcast add.

PiperOrigin-RevId: 296013376
Change-Id: Ie3ee2cc1569fd6df7bd94c63d5e919c1f4c98c5d

jvishnuvardhan committed:
Update Session.py

The current example provided in TF website throws an error in TF2.x as we need to disable eager to build a graph and run in session.
Please check the colab [gist here](https://colab.sandbox.google.com/gist/jvishnuvardhan/01e01bd71653f6566c68f5210fabdf65/untitled827.ipynb). Thanks!

Commit:
Use a TraceMe argument.

PiperOrigin-RevId: 296021551
Change-Id: Ifc86e30e5f7cb3972192137b7a92b647a63a2aac

tensorflower-gardener committed:
fix windows build from cl/295978584.

PiperOrigin-RevId: 296024920
Change-Id: Ib00b33130ff67d901487a27227d5c2599a8c3d7b

VoVAllen committed:
address comments

VoVAllen committed:
fix

allenlavoie committed:
Correctly record when an EagerOperation runs on a custom device

Should fix a performance regression where we looked up a CustomDevice each EagerExecute since we're now caching the lookup in EagerOperation (assuming the device name doesn't change and TFE_OpReset is used, like we do when executing from Python).

Also fixes a memory issue with custom device registration.

I still need to make TensorHandle::op_device_ a variant, but I think that can be split out into a separate change.

PiperOrigin-RevId: 296034610
Change-Id: I029b3b6927cd4efcf43beef8bd14ec50020eb089

nairb774 committed:
Make dump_graph safer to use on Windows.

Avoids using `\` in filenames, and uses JoinPath for constructing paths.

PiperOrigin-RevId: 296035763
Change-Id: I9e537af31b50b7eab3d120e0aa6a0a9de1381384

tensorflower-gardener committed:
make op_level_cost_estimator more robust with "random" input.

PiperOrigin-RevId: 296038412
Change-Id: I2a41674ffe8824bbdd5331af60264db7d10f198f

nairb774 committed:
Use GetTestUndeclaredOutputsDir to access TEST_UNDECLARED_OUTPUTS_DIR.

On Windows, Bazel populates environment variables with `/`s only. Changing path
manipulation logic to use `\` properly on Windows will conflict with this
behavior, requiring a layer of indirection to deal with Bazel.

PiperOrigin-RevId: 296038917
Change-Id: I9e6ae7492853966881db9c5fa53ced2383bce4aa

talumbau committed:
Armv8 asm fix: vector MOV

PiperOrigin-RevId: 296040987
Change-Id: I36b0e62ec2a95a6fa66644ce5bdc61f07fee168d

buchgr committed:
cuda_configure: fix quoting issue if paths contain spaces
PiperOrigin-RevId: 296041411
Change-Id: If3a679f013f6b44efd9739ac5b8eab169b52ab2a

advaitjain committed:
Use TF_LITE_REPORT_ERROR macro instead of error_reporter_->Report

PiperOrigin-RevId: 296042338
Change-Id: Ia4d99b29322fb9c5465854447770ed38697dbd2a

tensorflower-gardener committed:
Add pfor converter for StatelessIf.

PiperOrigin-RevId: 296043610
Change-Id: Ic91fc5c5d1cf44928bfbb8b0a13c7f304564c214

tensorflower-gardener committed:
Clean up redundant bazel bindings no longer used by grpc

PiperOrigin-RevId: 296044838
Change-Id: I34df2b59a5f02ac5da4ae1a5cd4de2054019f25f

tensorflower-gardener committed:
Removes a spurious check from AddRematerializedInstruction().

It is possible to rematerialize an instruction even if it defines a buffer that is being used by the instruction that is currently in progress. Even though that particular buffer does not help to reduce the memory usage, it may still be beneficial to rematerialize the instruction due to other buffers that it defines.

PiperOrigin-RevId: 296045046
Change-Id: I8236a04f420b341da9284b842484b815ccea4584

tensorflower-gardener committed:
Merge pull request #36578 from philpearl:new-tensor-perf

PiperOrigin-RevId: 296045986
Change-Id: I5b2e43d7c3c8b2c1edfe66dcab66361180549672

tensorflower-gardener committed:
Merge pull request #34870 from fizyr-forks:fix-tensorboard-with-keras

PiperOrigin-RevId: 296047937
Change-Id: I6b5710f6d666877a4cd9f8cd384bd6ff54d63e71

tensorflower-gardener committed:
make op_level_cost_estimator more robust with "random" input part 2

PiperOrigin-RevId: 296050162
Change-Id: I398aa4cfcf8bf4d007095e02800c84e20d1fc2bb

tensorflower-gardener committed:
Export public symbols for programmatic profiling APIs.

PiperOrigin-RevId: 296053405
Change-Id: I2d52a45e61bcb37d39d45be89c26a6a3f8f3ff1b

reedwm committed:
Run mixed precision tests in more cases.

PiperOrigin-RevId: 296053767
Change-Id: I4bcc64b9f09046b23cab0fd76e017f581242bfee

tensorflower-gardener committed:
Merge pull request #36743 from foldl:make-speech-command-v2-compat

PiperOrigin-RevId: 296056103
Change-Id: Ibee2e8731ff7e8ebf19e5fa4e53f62d95634e4b8

nairb774 committed:
Make use of GetDataDependencyFilepath and JoinPath to build paths which will
work across operating systems.

The previous implementation doesn't work correctly on Windows.

PiperOrigin-RevId: 296058125
Change-Id: I516774d9f45fb1b5f73d684416f98d91ab283266

tensorflower-gardener committed:
Merge pull request #36186 from punndcoder28:punndCoder28

PiperOrigin-RevId: 296058257
Change-Id: I837a5c6149e97b68cc5d85fe02c97a455486fb93

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 296061050
Change-Id: Ia8dfa50365e171f2deb0f22318cfbe75a1a1c9b1

tensorflower-gardener committed:
Adds additional error logging. Otherwise this ends up producing opaque messages like "Failed to evaluate the model.", when the problem is just label file doesn't exist.

PiperOrigin-RevId: 296061984
Change-Id: I394be0a1f6219879613ad11b794eefa9fb3d8dcd

tensorflower-gardener committed:
Change required numpy version to >= 1.16

PiperOrigin-RevId: 296062347
Change-Id: I416e4429606f130da6de08719da5693b82f0dcf7

tensorflower-gardener committed:
Export public symbols for on demand profiling APIs.

PiperOrigin-RevId: 296064710
Change-Id: I3a3b549fb59fe9ecbfc6c07ba809b0c1732932e4

buchgr committed:
Enable Remote Config for ROCM and CUDA RBE pre- and postsubmits

Previously TF_CUDA_CONFIG_REPO would point to a pregenerated and checked in configuration. This changes has it point to a remote repository intead that generates the configuration during the build for the specific docker image. All supported configurations can be found in third_party/toolchains/remote_config/configs.bzl. Each tensorflow_rbe_config() macro creates a few remote repositories to which to point the TF_*_CONFIG_REPO environment variables to. The remote repository names are prefixed with the macro's name. For example, tensorflow_rbe_config(name = "ubuntu") will create @ubuntu_config_python, @ubuntu_config_cuda, @ubuntu_config_nccl, etc.

This change also introduces the platform_configure. All this rule does is create a remote repository with a single platform target for the tensorflow_rbe_config(). This will make the platforms defined in //third_party/toolchains/BUILD obsolete once remote config is fully rolled out.

PiperOrigin-RevId: 296065144
Change-Id: Ia54beeb771b28846444e27a2023f70abbd9f6ad5

Commit:
Remove obsolete code. Replace the last places using it with the new version.

PiperOrigin-RevId: 296066537
Change-Id: I8f6748d9d2ea497822f4f65fa4301d2dd67db89c

Commit:
Add `shape_signature` to the Java & Python Tensor API.

PiperOrigin-RevId: 296067187
Change-Id: I2d98d92967cfe0429a9794685780d5f464b7882d

ckkuang committed:
Enable keras_save_load_test and increase shard_count to mitigate test timeout.

PiperOrigin-RevId: 296068534
Change-Id: I9dcf36cf9ebb78c5cd421dc442a7881bc5ddc232

nairb774 committed:
Make use of GetDataDependencyFilepath and JoinPath to build paths which will
work across operating systems.

The previous implementation doesn't work correctly on Windows.

PiperOrigin-RevId: 296070105
Change-Id: Ie4b5ecae64807682153dc17e6471984284edc111

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 296072123
Change-Id: Iae966a89cba24e82bb7c86ca30dda3581a0f98e8

mrry committed:
[Executor tracing] Add num_output_edges to the PropagateOutputs TraceMe.

This enables someone reading a trace to distinguish between long PropagateOutputs events that are due to contention, and long events that are due to a lot of propagation work going on.

PiperOrigin-RevId: 296072451
Change-Id: Ic7b731f3ccb2b0ba12e0e7d23f31e89cef9b0a97

tensorflower-gardener committed:
Merge pull request #36333 from alxhoff:conflicting_c_api_declaration

PiperOrigin-RevId: 296071818
Change-Id: Ia976b3dea901ae5b889da8dcd2017b0dd0a9acd5

Commit:
rebase

tensorflower-gardener committed:
Fix GitHub issue templates.

PiperOrigin-RevId: 296075106
Change-Id: I858fde7b2432be833c9524b9dff98d95cb55f96c

mrry committed:
[SparseTensor] Optimize the `tf.sparse.to_dense()` implementation.

This change includes several optimizations:

1. Introduce `SparseTensor::IndicesValidVectorFastPath()`, for validating the
   indices of a 1-D SparseTensor. The optimized code is similar to
   `IndicesValid32BitFastPath()`, which optimistically assumes that the tensor
   is valid and falls back to slower code in the failure case, except it does
   not have the 32-bit limitation. The compiler is able to vectorize the loop
   over the indices, for increased throughput.

2. Implement fast paths for 1-D and 2-D inputs in `SparseTensor::ToDense()`.
   The main win here comes from avoiding the data-dependent loop over
   dimensions when computing the index of the output value. We also avoid
   an unnecessary integer multiplication (by 1) in each case.

3. Minor optimizations to the 3+-D case in `SparseTensor::ToDense()`, avoiding
   unnecessary calls to `TensorShape::dim_size()` and using pointer arithmetic
   rather than Eigen logic to dereference index elements.

4. Minor optimizations to the `SparseTensor::Create()` method, which now
   assigns directly to the relevant fields of the result instead of invoking
   the `SparseTensor` constructor and the move assignment operator. In this
   case the existing move logic wasn't saving us much, because the `Tensor` and
   `gtl::InlinedVector` move constructors still have to copy quite a lot of
   data.

5. Minor optimizations to the `SparseToDense::Compute()` method. In particular,
   we avoid allocating a temporary tensor for the indices when the input is
   DT_INT64 (which is the common case, since all `tf.SparseTensor` objects have
   64-bit indices).

PiperOrigin-RevId: 296075159
Change-Id: I0b051621920aec9b2a8dc6c7ecbf55e5b2d59098

zhangyujing committed:
Strip default attributes before sending a remote RegisterFunction request, in order to support forward compatibility across RPCs.

PiperOrigin-RevId: 296075787
Change-Id: If536f03ab7d37fdba5d3431995d4b28d561ec78c

nairb774 committed:
Make use of GetDataDependencyFilepath and JoinPath to build paths which will
work across operating systems.

The previous implementation doesn't work correctly on Windows.

PiperOrigin-RevId: 296076015
Change-Id: I8c0876ab01bd1802657e51d9fc4b06271a2fea5c

Commit:
Allow operator invoke calls to return values other than kTfLiteOk and kTfLiteError.  Abort the invoke loop if a non-kTfLiteOk return value is found.

PiperOrigin-RevId: 296076951
Change-Id: Ibadc55d18e61231630cf82fdb6e5d283ad3a489d

qlzh727 committed:
Fix ConvLSTM2D layer with initial states.

Removing most of the duplicated logic and allow RNN layer to handle it.

Fix #35306

PiperOrigin-RevId: 296079716
Change-Id: I2a5506f7fad34405b0b2ff0dae13c14682e9a349

tensorflower-gardener committed:
Add MakeUnaryHlo() and MakeReverseHlo() to hlo_creation_utils.h/.cc

PiperOrigin-RevId: 296080049
Change-Id: I81d020a76da6820086a1a50379c77efc6c43918c

ezhulenev committed:
[TF:MLIR] Add canonicalization pattern to TransposeOp and compose a layout optimizer pipeline

PiperOrigin-RevId: 296081205
Change-Id: Ica9b311ba83e2e75b726eacbdc393c03692dacb8

tensorflower-gardener committed:
Merge pull request #36908 from tensorflow:jvishnuvardhan-patch-1

PiperOrigin-RevId: 296079738
Change-Id: Iab5914226bb3309b55b3e594ce1d09bc8ab900b9

nairb774 committed:
Avoid direct access to the env var TEST_UNDECLARED_OUTPUTS_DIR.

On Windows, Bazel populates this path with `/`s only making proper path
management impossible without sanitizing the path up front. This changes to
accessing the env var through an indirection layer which will fix path problems
on Windows when the codebase is ready to switch over.

PiperOrigin-RevId: 296083765
Change-Id: I26bbaf83ba5e3fafd3ab0a0de08f6cb597b94477

roserg committed:
Supported F32_F16 precision in Winograd transformations.

PiperOrigin-RevId: 296084853
Change-Id: If7f1715d84eae34159cf403d1ad208f9d1aa7305

roserg committed:
FullyConnectedTexture renamed to FullyConnected so as support all storage types.

PiperOrigin-RevId: 296085184
Change-Id: I3ea56947c7ddf70370c10b4375903880fd3d83c9

rxsang committed:
Make TPUStrategy work with tf.function(experimental_compile=True). This involves two changes:

1. Only create replicated var handle inside TPUReplicateContext.
2. If the function annotated with experimental_compile=True is called inside a XLAControlFlowContext, don't create a new XLAControlFlowContext.

PiperOrigin-RevId: 296086034
Change-Id: I821f3b3cd5ba69cd4c7bdb9c28e13e4b4c83f967

nairb774 committed:
Use io::JoinPath to build paths and avoid `\` when constructing file names.

JoinPath is being made to deal with different OS path separators. Defaulting to
`/` doesn't work in all cases. As an example, on Windows, when a `\` is in a
path, `/` is no longer considered a path separator which is why we want to avoid
it in filenames.

PiperOrigin-RevId: 296086151
Change-Id: Ib2dfef55e9e779ff5138f960dc462fce8a14833b

cheshire committed:
[XLA/GPU] Change rounding scheme for tree reduction to round up to nearest square

Previously, we were rounding up to the nearest divisor of the largest batch we
could handle without introducing atomics.
That leads to:
 - Very large padding, e.g. rounding up 8193 to 16384
 - Very small dimensions of extra reduction kernels, e.g. 2

Instead, this CL uses a more "even" rounding scheme, where we round up the
number to the nearest square. Nearest square is guaranteed to be within
2 * sqrt(N) of a number N, so required padding is fairly small even in the
worst case.

PiperOrigin-RevId: 296086172
Change-Id: I7bfa72b2309fd1e3c596d6e028a9468660f84879

MarkDaoust committed:
Remove useless self-link and python version note.

Fixes #26645

PiperOrigin-RevId: 296086404
Change-Id: Ice56c6032290939e89fd752d584baab3d320c689

smit-hinsu committed:
Automated rollback of commit e623eb0f9c1c65705f0cfb1c6cb1d8cb2649cdbb

PiperOrigin-RevId: 296088251
Change-Id: If0555f6a1f01eb03fda7dc33377bfdb317843740

crccw committed:
Change to use assertLen in one test

due to pylint complaints

PiperOrigin-RevId: 296090392
Change-Id: I4faf420c108b011154b3f4a7c3eee68482ef9de4

tensorflower-gardener committed:
Fix a bug in ctc_loss_dense with unique.

1. Unique label has to consider the case where blank label is not 0.

2. The scattering mechanism assumes that 0.0 always corresponds to a padding region, but this is not the case when there is only single valid path (0.0 = log(1.0)). This happen when the lengths of the logits and the label are the same.

PiperOrigin-RevId: 296090785
Change-Id: I803508252e688571bca531b1aa95dd2160902d4c

omalleyt12 committed:
Keras ideal fit and compile.

Kept all new abstractions private for now. In a few weeks, if we're
comfortable that these abstractions are working and stable, we should expose
many of them publicly.

Capabilites added by this CL:

(1) Easy to create a custom training step via overriding Model._train_step
(2) Easy to create custom tf.function / DistStrat logic via overriding
Model._make_train_function
(3) Advanced users can override Model.compile and Model.fit
(4) Full support for dicts, nested structures, etc with Subclassed Models.
(5) "Power user" path (tf.data inputs) only modifies data in Model._train_step,
where this behavior is easy to override and disable. This applies even to
Keras's assumption that data is passed in (x, y, sample_weight) format.

Behavior changes:

(1) "loss" passed to Callbacks is now stateful (like all other metrics in
Callbacks). This greatly simplifies the training step logic and callback logic.
(2) ProgbarLogger always uses steps. If steps is not available, the
ProgbarLogger handles inferring the steps after the first epoch.
(3) validation_batch_size added in `fit`, rather than inferring from generator.
(4) Model.inputs, Model.outputs, Model.input_names, and Model.output_names are
no longer populated for subclassed Models. Instead, "pseudo" output names are
created for subclassed Models, which are only used for metrics names and
SavedModel's signature.
(5) Cast NumPy floats to backend.floatx(), otherwise leave
unchanged (this is likely not a change, we did something like this in our old
version but the logic was scattered in many places)

PiperOrigin-RevId: 296090972
Change-Id: Ia5ac833fd39085bddb016833bd338083d0dc5fc2

lrdxgm committed:
SVDF: Do the activation-state-shifting at the beginning of each Eval, sparing an extra reinitialization of the latest activation.

Note this does change behavior: the state tensor will be shifted in time - or more precisely, it will not be shifted after executing a single step.

PiperOrigin-RevId: 296091378
Change-Id: I24cd2bf0ece80d524a31271db61f6bcbdc40b5b9

smit-hinsu committed:
Automated rollback of commit 9f86c8c5a42e51b42880b69cde7f43f60d7276cc

PiperOrigin-RevId: 296092477
Change-Id: I12be08294a71f885a67760098b732a47ff595384

tanzhenyu committed:
Error out when saving IndexLookup layer.

PiperOrigin-RevId: 296093473
Change-Id: I96ce0319a1480399c47d10ecc048007483eca595

liufengdb committed:
Fuse hardwish for mobilenet v3

The mobilenet v3 frozen graph has extra FakeQuant ops which blocks the fusion,
thus we create a special pattern to remove the redundant FakeQuant ops.

PiperOrigin-RevId: 296093529
Change-Id: Ic5bc6808afb12b2004ed7b6f3a81f914df917d5e

nairb774 committed:
Use Env::LocalTempFilename for a temp filename.

This function works both in and outside of tests. Additionally,
LocalTempFilename works well on Windows where as TmpDir is a little problematic
because of bazel oddities.

PiperOrigin-RevId: 296093672
Change-Id: I998ac9dab0077d7cfc09631db4bf295f4eef155a

tensorflower-gardener committed:
Make map_xla_to_scalar_op as a library.

PiperOrigin-RevId: 296094105
Change-Id: I584fceb8adafded0b44b0fc0fb5acb336d6a35d2

cheshire committed:
[XLA] Respect TF_DETERMINISTIC_OPS environment variable for reductions

PiperOrigin-RevId: 296094275
Change-Id: Iadcbf33d5d6432413c86d4d176865980de252eeb

smit-hinsu committed:
NFC: Add a TODO to move HLO Relu legalizations to TF to TF lowering
PiperOrigin-RevId: 296095163
Change-Id: Ic0b8d26c11c64e6584eef0da38b87e71d2dd03e8

impjdi committed:
Remove duplicate error check in RESHAPE.

PiperOrigin-RevId: 296095268
Change-Id: I22c50cf2fb8cb1343c8c99f9cf557aad45aff15c

nairb774 committed:
Use JoinPath over a fixed string for building paths.

The fixed path doesn't work well on Windows when the correct path separator is used.

PiperOrigin-RevId: 296095586
Change-Id: I9fe0459ef58a310bf471cf2548b3f7e23b764502

crccw committed:
Retire agg_small_grads_max_bytes, agg_small_grads_max_group

agg_small_grads_max_bytes and agg_small_grads_max_group aren't effective to public users. They're not exposed in the public API, so they're always their default value (agg_small_grads_max_bytes=0), which are then ignored.

PiperOrigin-RevId: 296096179
Change-Id: Id7397539441a0e34af5c76d994eb08028e289b6d

allenlavoie committed:
Add an experimental eager C API for generically fetching and setting op attributes.

Right now you can only fetch the whole attribute map and set it wholesale, but we can add more fine-grained attribute control in the future.

This allows the custom device API to pass in attributes, and custom devices to forward these to their own TFE_Execute calls. This is required for creating variables.

PiperOrigin-RevId: 296096192
Change-Id: I98c23bdcd13e479235b3e27850b1bb0bd7a53bba

Commit:
Fix Bazel not building anymore with the commit 09fe958f

Commit:
Allow user to pass input_shape to split.

PiperOrigin-RevId: 296099673
Change-Id: I2ea990d2e91a991fb1a89cf7ec5f1c749caaec5b

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 296102169
Change-Id: I07271901a4d49284d377b3d0da6fbb5cb1aeef27

tensorflower-gardener committed:
Fix some recommendations in the Profiler Overview Page.

PiperOrigin-RevId: 296103334
Change-Id: Ia9a0aff4044d77b8e8f91bc4a953325b47b28c9d

nairb774 committed:
Use GetTestUndeclaredOutputsDir to access TEST_UNDECLARED_OUTPUTS_DIR.

On Windows, Bazel populates environment variables with `/`s only. Changing path
manipulation logic to use `\` properly on Windows will conflict with this
behavior, requiring a layer of indirection to deal with Bazel.

PiperOrigin-RevId: 296104352
Change-Id: Ibaa19d0c4d231a15811232c63bcefc9d4931f88b

ezhulenev committed:
[TF:MLIR] Add operation interface for folding operands transposes into the ops

PiperOrigin-RevId: 296105420
Change-Id: Ie8c54de100910f6eda53bf2d02b194a0a8785ec8

guizili0 committed:
Merge branch 'master' into guizili/dequantize

Commit:
Add verifier for HLO Iota op.

Also fixes a bug in tf.RandomShuffle legalization caught by verifier.

PiperOrigin-RevId: 296109247
Change-Id: Icea818f51a6eab91f65efb65aa07f9639d9704a6

advaitjain committed:
Remove unnecessary TF_LITE_MICRO_TENSORS_PREPARED.

PiperOrigin-RevId: 296111850
Change-Id: Ie817bad07ad60b12ecd05aaa82de7f03b476972c

Commit:
Fix TPU nightly build script.

We can't pass a single string containing all the arguments, since it'll be taken as a single argument instead. Storing as an array allows us to safely expand.

PiperOrigin-RevId: 296112909
Change-Id: Ic22b58bd8e6f9bbeadcbec5e0a78c2ac2e122f9c

wangtz committed:
TFL: slightly speed up reference::Softmax by avoiding unnecessary float->double cast.

The original logic is a bit weird that the calculation is in double while the accumulator is in float. Also in general, beta doesn't have a huge significant figures.

PiperOrigin-RevId: 296114624
Change-Id: I4e43fb9606b7b3c9f352de46da5d36cc50d7897a

nhasabni committed:
Merge branch 'master' into nhasabni/dnnl1x_matmul

crccw committed:
Add aggregation to OptimizerV2.apply_gradients

This option allows post processing of all reduced gradients, without inheriting from optimizer.

PiperOrigin-RevId: 296118658
Change-Id: Ifb6884ec981b06eb70fe5ee9126ab9ac013550e9

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 296121312
Change-Id: I8eefa17cfa266fb6d642381d63ecfc0c6ffa0ba0

tensorflower-gardener committed:
Merge pull request #36738 from Intel-tensorflow:nhasabni/dnnl1x_matmul

PiperOrigin-RevId: 296125627
Change-Id: Ide2ab5a6f9ba19b72c65f3eb645a24caa34ec7e5

hawkinsp committed:
[XLA:GPU] Add an AllReduceCombiner pass, that merges AllReduce operations.

On GPU, implement combined allreduces using NCCL groups.

PiperOrigin-RevId: 296130269
Change-Id: I763f0139c8ed9a59d7d691e3252e6b46244fefd6

renjie-liu committed:
In verify ophint extraction: instead of error out, we should just don't do anything.

PiperOrigin-RevId: 296135220
Change-Id: Id67e57859bc73ba13c4844e73691fcdff37894c1

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 296135428
Change-Id: Icbe9c3556c2c8bcdf0328e932ff73eeed95dd2da

tensorflower-gardener committed:
Add pfor converter for "If"

PiperOrigin-RevId: 296138673
Change-Id: Ib63ecece09d8d1df4e53c69dae0b511aaba6c120

srvasude committed:
Add test for Expm1 for small parameter regime of complex numbers.

PiperOrigin-RevId: 296138822
Change-Id: Idf7ff8e34acb056bda59002f54b6d3df7c42ba5a

jaingaurav committed:
Use an AbstractOperationInterface in TFE_Op

This allows us to move towards cleaning up some of the header
dependencies in incurred by c_api_internal.h.

PiperOrigin-RevId: 296139196
Change-Id: I02be01d98ad06af8f1f2a8fc1f067849336a0c26

renjie-liu committed:
Add function to load saved model for tflite mlir converter.

PiperOrigin-RevId: 296139934
Change-Id: I1c608c2971d81e5efa38925ee9fe4b80f437726a

renjie-liu committed:
Legalize ophint converted node in mlir. (this is the first cl of a series of cls for import ophint python directly converted model).

PiperOrigin-RevId: 296140203
Change-Id: I726b4b88d7fe7878c283f7806b15538304fba7f9

tensorflower-gardener committed:
Merge pull request #36742 from Intel-tensorflow:guizili/dequantize

PiperOrigin-RevId: 296140320
Change-Id: I1e1baee553e97bc3de9d5cd58d8af15d38a81341

tensorflower-gardener committed:
Merge pull request #36744 from Intel-tensorflow:pallavig/dnnl1x_concat

PiperOrigin-RevId: 296140331
Change-Id: Ic34558bfc4848979ad26b6b38b8eea630f865205

blakehechtman committed:
[XLA] Avoid hash collisions in CseHash.

PiperOrigin-RevId: 296143190
Change-Id: I16cef346311b419f04911c241462fa55a5aa04ad

terryheo committed:
Add Maximum & Minimum op support for GPU delegate

Refactored elementwise op kernel to handle Maximum & Minimum.

PiperOrigin-RevId: 296146084
Change-Id: Iefd333b79638d8705b28167657af475aa75e639a

terryheo committed:
Add Maximum & Minimum op support for Metal

PiperOrigin-RevId: 296149175
Change-Id: I3d26f756cb8f5fe0d94fac3f8515da8b2124dcc4

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 296149237
Change-Id: I9d0bdb02115df83422421306024dd5255a320768

akuegel committed:
Add a __init__.py file for tensorflow/compiler/tests.

This is recommended standard, even though it looks like
bazel automatically adds this if it is missing.

PiperOrigin-RevId: 296150499
Change-Id: Iadf194be068d81aa07f4365b1057f3b7c28a6190

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2020-02-20

PiperOrigin-RevId: 296151051
Change-Id: I570dbed58cefe274637462ca4d160abf36fac313

buchgr committed:
preconfig: remove checked in rocm and cuda10.0 configs as these configurations are placed by remote config
PiperOrigin-RevId: 296152981
Change-Id: Ib19efb421a63a3b2e87d862eda3a34816e8a3890

terryheo committed:
Fix iOS build failure

Removed absl::StrCat() usage.

PiperOrigin-RevId: 296159687
Change-Id: I1f825cbda293c7e218921563c3602777b9e29810

tensorflower-gardener committed:
Merge pull request #36904 from vinceab:master_fix-benchmark-build

PiperOrigin-RevId: 296172936
Change-Id: I553089a37d7856ca56233bc8d50c6529678ad4fa

nikochiko committed:
Single quotes -> double quotes for LazyLoader

nikochiko committed:
Make pylint comments consistent

Commit:
Implement lowering of dynamic_broadcast_in_dim from HLO to LHLO.

PiperOrigin-RevId: 296198208
Change-Id: Iabe355fd8f87545dac6c9b537682668a2c653afb

tensorflower-gardener committed:
Merge pull request #36356 from rahul003:s3_ca_paths

PiperOrigin-RevId: 296212514
Change-Id: I1d51cbad154b4f8d6df27ad7e3736c39c93c2863

tensorflower-gardener committed:
Merge pull request #36210 from rrkarim:master

PiperOrigin-RevId: 296213895
Change-Id: Ie38d58befbff44fcaa8af435dae356cb31b55d72

Commit:
Support materializing dynamic broadcast operations for binary operations.

This inserts a xla_hlo.dynamic_broadcast_in_dim operation for the two operands of the dynamic operation of the rank is known and a broadcast_dimensions attribute is present.

PiperOrigin-RevId: 296216205
Change-Id: Ic5e5d80ce5921be91dd6c023af32a402859f24f4

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 296220963
Change-Id: I740d8c4b671dc71c987dcd9c81150f8838a64f33

jaingaurav committed:
Reduce overhead of protecting tensors for eager

The eager executor tried to prevent forwarding of any input tensors by
incrementing the reference count of any "non-consumed" inputs. This
involved highly delicate logic which first signaled "non-consumed"
inputs as those with a reference count greater than 1 (1 from python and
another from the EagerOperation class), which require "protecting" by
incrementing underlying tensor buffer. This logic is highly heavyweight
for the common case of synchronous execution. We thus simplify the logic
by having all TensorHandle Tensors protected at construction and
"unprotect" then if the reference count is 1.

- Hold 2 reference counts a TensorHandle's backing Tensor. This protects
  the Tensor from being forwarded.
- Add the ability to unprotect a TensorHandle's backing Tensor when the
  reference count is 1.
- Split ExecuteNode into Async implementation. The sync ExecuteNode
  class can avoid various copies such as the list of inputs and the
  forwarding map.
- Remove the experimental TFE_OpConsumeInput API. Input forwarding can
  be achieved by releasing the handle after calling TFE_OpAddInput as
  demonstrated by the added tests.
- Fix TF_AllocateTensor to return a forwardable tensor it was previously
  disabled due to re-using the logic in TF_NewTensor.
- Save mirror tensor when calling TFE_TensorHandleResolve.

PiperOrigin-RevId: 296225251
Change-Id: I484cfccbef8b44e82757b8bda0981cd7fd2f8096

tensorflower-gardener committed:
Merge pull request #36891 from orf:patch-1

PiperOrigin-RevId: 296225274
Change-Id: Ie14fd3afc7f602e9c950cee8761f0dd341755a95

tensorflower-gardener committed:
Merge pull request #36886 from Officium:master

PiperOrigin-RevId: 296225695
Change-Id: I85cc9bf4993a11eaa1ca6633020008890be0d975

omalleyt12 committed:
Revert Dense layer changes that attempt to support 1d inputs.

PiperOrigin-RevId: 296232311
Change-Id: Iffc4ea2267b8846dec15397328ef9bf1ddc21760

liyunlu0618 committed:
Internal change only.

PiperOrigin-RevId: 296232961
Change-Id: I083e446349c1c68b4a21fe58d59a6babe93e69f2

tensorflower-gardener committed:
Remove spurious std:cerr debugging statement.

PiperOrigin-RevId: 296234228
Change-Id: I4817dbcaf48ab37fe68df18bad5f030746099341

MarkDaoust committed:
Add a `classifier_activation` option to keras.applications

Defaults to `softmax` (the current behavior) but now users have the option of turning it off, or setting a different activation function.

PiperOrigin-RevId: 296235461
Change-Id: Iea01b8a2b260ece5e91b4dc1e6e42526136a2066

MarkDaoust committed:
Fix doc generator to handle new package layout.

tensorflow_core is gone in tf-nightly.

PiperOrigin-RevId: 296238681
Change-Id: I604be239c807b6e6fb9569560d9f94326b303711

nairb774 committed:
Use Env::LocalTempFilename for a temp filename.

This function works both in and outside of tests. Additionally,
LocalTempFilename works well on Windows where as TmpDir is a little problematic
because of bazel oddities.

PiperOrigin-RevId: 296239768
Change-Id: Ie1c44de9f4a0b31100ec66979152c39a5e2a965f

nairb774 committed:
Use Env::LocalTempFilename for a temp filename.

This function works both in and outside of tests. Additionally,
LocalTempFilename works well on Windows where as TmpDir is a little problematic
because of bazel oddities.

PiperOrigin-RevId: 296239824
Change-Id: I4e636bf150fc5554503e14361a5953598c9db638

tensorflower-gardener committed:
Merge pull request #36502 from 372046933:patch-2

PiperOrigin-RevId: 296239434
Change-Id: Ief931e94db1e71a231073187df597007d9a7e8a6

tensorflower-gardener committed:
give sendop/recvop some temporary attribute for tracing. so we can annotate the memcpy device events better.

PiperOrigin-RevId: 296242264
Change-Id: Ib515bc56faf37ede6610b62c8aaab3ab66ef6830

nairb774 committed:
Add comment flagging people towards Env::LocalTempFilename instead of testing::TmpDir

PiperOrigin-RevId: 296243323
Change-Id: I544c24625d36c5b72c04964c0b9ca5ed3a44fa4c

tensorflower-gardener committed:
Merge pull request #36876 from wuhy08:add-tflite-quant-leaky-relu

PiperOrigin-RevId: 296244927
Change-Id: Ibfc7231d823257a88f6ba05e2df1604c5530383f

tensorflower-gardener committed:
Merge pull request #36856 from wwwind:crash_conv_dilation

PiperOrigin-RevId: 296249506
Change-Id: Ic6401824fab91257354e8ae3dfbbdba505304f3c

nairb774 committed:
Use Env::LocalTempFilename for a temp filename.

This function works both in and outside of tests. Additionally,
LocalTempFilename works well on Windows where as TmpDir is a little problematic
because of bazel oddities.

PiperOrigin-RevId: 296250888
Change-Id: I2a8bc52ad784eda4d00f63c91eec681cc91e16e7

nairb774 committed:
Use Env::LocalTempFilename for a temp filename.

This function works both in and outside of tests. Additionally,
LocalTempFilename works well on Windows where as TmpDir is a little problematic
because of bazel oddities.

PiperOrigin-RevId: 296250909
Change-Id: I313e4e3467e8f5956c681adb577c70918fe853b6

nairb774 committed:
Make use of JoinPath to build paths to path references can work correctly across operating systems.

PiperOrigin-RevId: 296251146
Change-Id: I4db57dfb924ded085d2cb20969193e497100c052

nairb774 committed:
Make use of GetDataDependencyFilepath and JoinPath to build paths which will
work across operating systems.

The previous implementation doesn't work correctly on Windows.

PiperOrigin-RevId: 296251590
Change-Id: Iac5ad8dbf78d06969a51b9476f66e0b8affdaaa4

yifeif committed:
Remove the main repo reference.

PiperOrigin-RevId: 296251652
Change-Id: Ide83f2957a08b688838d8de7af92fd1cc36369e5

nairb774 committed:
Use Env::LocalTempFilename for a temp filename.

This function works both in and outside of tests. Additionally,
LocalTempFilename works well on Windows where as TmpDir is a little problematic
because of bazel oddities.

PiperOrigin-RevId: 296251680
Change-Id: I985f178e2e85105cf79c4572b9158e168490348c

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 296254153
Change-Id: I475e330a8465070c9d6ee6789f46d0e1ccb9658f

gunan committed:
Make sure aot codegen test is not confused if windows introduces CRLF line endings.

PiperOrigin-RevId: 296255135
Change-Id: I234fe6b76f0cd9ead1d3dc69bf657160d2d910f8

nairb774 committed:
Remove recursive visibility restriction subsumed by broader restriction.

PiperOrigin-RevId: 296259183
Change-Id: Iadb880dae371a95ba8f51e2f0b34d2445dbf1ff6

tensorflower-gardener committed:
Merge pull request #36913 from DEKHTIARJonathan:patch-1

PiperOrigin-RevId: 296255144
Change-Id: I55760f36937576db51a4eaf6c555be7cb46bce5a

aaudiber committed:
Fix bug where MapAndBatch fusion reverts parallelism to 1.

To be affected by this bug, a user would need to either manually set their forward compatibility window past 3/6, or set the recently-added deterministic argument in a call to Dataset.map().

PiperOrigin-RevId: 296263821
Change-Id: I4414719f53d80364503880b784b2389c099dc62b

tensorflower-gardener committed:
Override buffer size when running on cloud tpu.
A small buffer size adds a performance slowdown. Here we override the buffer size to a minimum recommended buffer size.

PiperOrigin-RevId: 296265141
Change-Id: I3a9ca24d9a89a810407ce87cc33bc6f4540ce47a

haoyuz committed:
Automated rollback of commit a7c9317a40f85bdfd606d10a6f8f3d21325d0f95

PiperOrigin-RevId: 296265367
Change-Id: Ie631071ec9894f297db3f82d43153be11ec249c9

Commit:
NFC: Use OpRewritePattern in xla_hlo to std legalization.
PiperOrigin-RevId: 296269235
Change-Id: I8a58df72db49531993e384e041d2d7f5a14648bb

omalleyt12 committed:
Add tests to make sure optimizer weights are being saved correctly.

PiperOrigin-RevId: 296269488
Change-Id: I714119a593c8846faf253aa94ef060b141e4ad84

hyouklee committed:
Replace replicated cross-replica AR with global AR with division

PiperOrigin-RevId: 296271095
Change-Id: I19aed2a6ef74cd26d18151b449e4fe98d34ce8ba

crccw committed:
Test both CollectiveAllReduce NCCL and RING

PiperOrigin-RevId: 296274147
Change-Id: I61f6f445bed2829707f5f7126d0a1f765227496c

jsimsa committed:
[tf.data] Add pre-requisite check for `padded_batch`.

PiperOrigin-RevId: 296276337
Change-Id: I068975d1320e688a2826114137149f66ab37b4e0

nairb774 committed:
Formatting changes to support tooling.

PiperOrigin-RevId: 296279985
Change-Id: I3b983f245ac4ede32bceac8c609b521a4a7fc22b

tensorflower-gardener committed:
Fix eager:core_test test indeterminism.

One of the tests in the suite left the context inside a CPU device
context. This made other tests fail depending on the order in which
they were executed. It also caused some tests to /pass/ because they
required that device context to be (accidentally) active.

PiperOrigin-RevId: 296281260
Change-Id: I6a2dcb4bd566591c3d198d1e2b2c2545af108916

nairb774 committed:
Formatting changes to make tooling easier.

PiperOrigin-RevId: 296281482
Change-Id: I736d812a7fcdc05695367776cb4177649c69cffc

pifon2a committed:
[MLIR] Use PassOptions in LhloFuseLinalg pass instead of LLVM CL opts.

PiperOrigin-RevId: 296282382
Change-Id: I1c4223c40e8816259984bd2971c4f535e3735830

Commit:
Move Iota folding to xla_hlo-std legalization.

Iota op folding can lead to huge constants. Not every target would like to fold iota ops which increases the file size. Moving it to xla_hlo to standard legalization which was the original intent behind adding this fold.

PiperOrigin-RevId: 296282985
Change-Id: I71cb1679796ff0d36251ddb2f4cd0fce8aa75192

nairb774 committed:
Use resource_loader to reference in-tree resources.

PiperOrigin-RevId: 296283401
Change-Id: I8531d318a672c1d496f334932ec7355b1d343adf

tensorflower-gardener committed:
Merge pull request #36852 from tfeher:allow_build_at_runtime

PiperOrigin-RevId: 296283355
Change-Id: I1161707e70e26c1f348104958bbc08cd6354e7b2

smit-hinsu committed:
Temporarily disable mkl_dequantize_op_test

PiperOrigin-RevId: 296287811
Change-Id: Icc89d8cf863eef95ba3e4c1b7098bcb6327a9494

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.
PiperOrigin-RevId: 296294694
Change-Id: I57dcfe38ff03af3cf7987554f5553b66f04716db

nairb774 committed:
Make use of GetDataDependencyFilepath and JoinPath to build paths which will
work across operating systems.

The previous implementation doesn't work correctly on Windows.

PiperOrigin-RevId: 296295721
Change-Id: I1d4d067a5c938cfd6c1ce8724bb9f49ea89a4bda

aaudiber committed:
Fix error string.

PiperOrigin-RevId: 296299506
Change-Id: I17388239c97ccb4ece0fe04c33e0a650f185670f

ezhulenev committed:
[TF:MLIR] Make Conv2D layout sensitive operation and update LayoutSensitiveInterface

PiperOrigin-RevId: 296299559
Change-Id: I14810117004b724f0d054885a9d4fef45195128a

advaitjain committed:
Stub out TFLITE_ASSERT_FALSE for NDEBUG builds.

Also,
 * add custom debug_log for xtensa-xpg that will be empty for NDEBUG builds.
 * Linux builds via the Makefile now do not have -DNDEBUG to be consistent with
   the bazel builds.

PiperOrigin-RevId: 296300474
Change-Id: Ia1473e23bd8705f520beace7ee704479b0c52117

qlzh727 committed:
Fix the cache key problem when compute_output_shape().

This is a very tricky one wrt the id() of int in python. Under the hood,
id returns memory address for the int, and python has a cache location
for the ints, which result into different ints get same hash value.

Changed to use tuples of shape itself as the dict key, since the tuple
itself is immutable and hashable. Same tuple value will return the same
hash value.

Also remove the generic utils for that where network.py is only usage
for that function.

Fix #32029

PiperOrigin-RevId: 296302946
Change-Id: I865c9380a06ed6ee80fea7f942c21c4d102473c2

tensorflower-gardener committed:
Suppress 'conversion to a dense matrix' warning from LinearOperatorFullMatrix.solve().

The current warning is inappropriate: since a LinearOperatorFullMatrix is inherently dense, no efficiency is lost when we treat it as dense.

PiperOrigin-RevId: 296305093
Change-Id: Id3b7e2a00f05d1e516374c4241cd84529844a056

Commit:
Stop the gradient for QuantizeAndDequantizeV2 when the input is out of range.

PiperOrigin-RevId: 296306551
Change-Id: Idcc1153ed7bfcac6cd9b6533800bf753d6ec166e

nairb774 committed:
Automated rollback of commit 5fc1ad961b83dd36941aa2b447a4a602b622e2c9

PiperOrigin-RevId: 296307212
Change-Id: I8ab232f276fafe6302bcd9f67d805235ab083ec3

tensorflower-gardener committed:
Merge pull request #34388 from nikochiko:fix-save-model

PiperOrigin-RevId: 296307484
Change-Id: Ic98ef36bb585f9c07d6522bf671954ff5f3bbad2

gunan committed:
Set the worker pool name constraints for bazel windows rbe toolchains.

PiperOrigin-RevId: 296309660
Change-Id: I5327f5c5005e9959e0058832a5d0cb710475d7fe

Commit:
Update the Dockerfiles for manylinux2010 toolchains with Python3.8 installations.

PiperOrigin-RevId: 296311657
Change-Id: Ic2040763c882ff3a1295100a8ec9dbf4fea4c50d

henrytangoogle committed:
Call InitializeNumpyAPIForType() to make bfloat16 happy.

PiperOrigin-RevId: 296313339
Change-Id: If4f61d7de9b94860107f91e39fedc05d2807538d

tensorflower-gardener committed:
Add a pass that lowers regions within tf_device.parallel_execute op to islands. This will ensure that computation in multiple logical cores will be executed concurrently.

PiperOrigin-RevId: 296314462
Change-Id: I39c22e52ad1aa813ee0c31dffd6d399461b5c04b

tensorflower-gardener committed:
tf.Benchmark: add mean and stdev of the wall time to unreported extras.

PiperOrigin-RevId: 296316355
Change-Id: I2e2de5de0b92c06c6b3d677ca5ea57046e0209b0

tensorflower-gardener committed:
Allow using tf.identity to explicitly copy tensors.

When device placement policy is 'explicit', and a tensor copy is
required for an operation input, the program will fail with an error
message that suggests a solution of the form

  with tf.device(dev): x = tf.identity(x)

This does not work currently. This change adds a policy exception for
the "Identity" operation, so the above starts working.

PiperOrigin-RevId: 296317942
Change-Id: Ie2e79c8b095bed9376976a8dcd1b49a0af9b7dc9

allenlavoie committed:
Forwardprop: fix nested forwardprop of non-differentiable ops

A special case was expecting that we didn't have any tangents, when in fact we just want to discard them.

PiperOrigin-RevId: 296318800
Change-Id: I7198596435f294333e00a2dcfe5ac8ec31d0b28c

jianlijianli committed:
Add another integer LSTM reference runtime for special hardware platform. Its quantization will be added separately.

PiperOrigin-RevId: 296320108
Change-Id: Ifb3a22667322a5af49426b5a1fc4066a50beac51

tanzhenyu committed:
Automated rollback of commit 0d2f3be5ebe4c762dddad2fe1bac1b4af538de2c

PiperOrigin-RevId: 296320816
Change-Id: Ib8b5857178fa10513755de65ffcde1adf6dabad3

impjdi committed:
TFLite GPU: Implement SPACE_TO_DEPTH.

PiperOrigin-RevId: 296321368
Change-Id: I3b5844fde83ef48002a4c326eeb745587068c208

Maratyszcza committed:
Fix typo in validation of beta parameter in XNNPACK Softmax

PiperOrigin-RevId: 296321710
Change-Id: I3449cf2809ddb72bc17081725797b0699d33d8a3

smit-hinsu committed:
Temporarily disable framework_memory_checker_test with tsan and msan

PiperOrigin-RevId: 296323364
Change-Id: I2f98fa7716925140b69ba3cb2c5cb4d94fd57cdf

fchollet committed:
Add dataset_from_directory utility.

PiperOrigin-RevId: 296323743
Change-Id: Ia6a6824e8f9bd45d7ac726e7c21477f8d85540ab

roserg committed:
Fixed mean work group size for Adreno3xx.

PiperOrigin-RevId: 296323931
Change-Id: If3555991cb399a9e3f850541276ec8d132c20e37

roserg committed:
Improved workgroup selection for Winograd transformations.

PiperOrigin-RevId: 296325182
Change-Id: Id8a593564142316cfdff3b5c9357af11358a6c19

hawkinsp committed:
[XLA:Python] Add TopK operation to Python API.

PiperOrigin-RevId: 296327503
Change-Id: I345150c480b48ba97645376674faa6109f6631a7

rchao committed:
Supplement multi_worker_callback_tf2_test.py with tests that have the same file path for ModelCheckpoint and TensorBoard callbacks.

PiperOrigin-RevId: 296328002
Change-Id: I28bb1d4b60e1fb47c1570852fe82d71adc6ebffe

nhasabni committed:
[Intel MKL] Fixing Device() API usage in mkl_eager_op_rewrite.cc

API change in TF core caused a build failure in MKL CPU backend. This commit fixes the compilation failure.

haozha111 committed:
Update minimum op version for TF 2.2.0 branch cut.

PiperOrigin-RevId: 296328883
Change-Id: I3deda696e7ad2c35cbd580decd72ca79e91963e4

roserg committed:
Added flag for separate weights for H dimension in ConvTexture.
This flags allows to execute convolution for winograd transformed input.

PiperOrigin-RevId: 296329659
Change-Id: I97b87f7b7f69b11452173e1e450b59542befa3cb

ezhulenev committed:
[TFRT] Add layout optimization pipeline to tf_to_corert

PiperOrigin-RevId: 296330081
Change-Id: I641a1677896e83b4b1b05d6d7688bb80c8dbeb8b

jpienaar committed:
Pack should verify compatible rather than exact type

Unknown types should not be flagged as illegal there as they could resolve to
valid types at runtime and so need not be illegal.

PiperOrigin-RevId: 296330175
Change-Id: I2664fd817b223f89d54e3301b263bfa770d7d24e

lrdxgm committed:
Remove unnecessary returns.

PiperOrigin-RevId: 296330462
Change-Id: I7e88ff943d02a160353b30d313a62e266bcad4cf

reedwm committed:
Do not crash if losses have different dtypes.

Also change mixed precision tests to no longer cast losses to float32, as this is no longer necessary, even when regularizers are used (which always are float32)

PiperOrigin-RevId: 296333364
Change-Id: I2f026815b6b71a701ad3975654cef1fe26ea0728

jpienaar committed:
Restore the docs directory that was accidentally removed.

PiperOrigin-RevId: 296334991
Change-Id: I6e2fe1a8d5f3e194dcd381e58d96bbb28c45580c

Commit:
Set shard count in the input test to prevent timeouts

PiperOrigin-RevId: 296336666
Change-Id: I04bc3e046f18cd0a72c891d65be34298bf16c202

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 296339143
Change-Id: Ie9f9b914c9f3b660eafc9fad080d7935ee0466b1

miaout17 committed:
Enable TFLite experimental new converter by default.

PiperOrigin-RevId: 296339261
Change-Id: Ibae109e8ebc3dae196e144b6ec7a740d4b7c82fd

tensorflower-gardener committed:
Internal change

PiperOrigin-RevId: 296339357
Change-Id: Ife4d6cc532586e15b94c049786977c4a7acf597d

teijeong committed:
properly initialize shape_tensor in Hexagon delegate reshape

PiperOrigin-RevId: 296341583
Change-Id: If3b36d6941d85494ea73fa8ff4207152d6cf48d5

xhae committed:
String input support on TFLite Tile op

PiperOrigin-RevId: 296342758
Change-Id: I4498b56b6da7074f8747fab893009b0d1d0d3cc9

wangtz committed:
TFLM: Reduce the latency for Reshape operator.
This is achieved by moving shape check to prepare so that it's ran only once.

PiperOrigin-RevId: 296343354
Change-Id: Ie72628b5abf8cc949dd4c8d1190007bab5f0ff1e

mgester committed:
Fixed control dependency errors in BreakUpIslands

Previously, some necessary control dependencies after breaking up islands were
missed (e.g., a dependency between a newly created island and a SwitchNOp), and
"Adding control dependency not supported" errors were reported in such cases.
Fixed this and added tests that contain all previously problematic ops and check
that control dependencies are now correctly added.

PiperOrigin-RevId: 296343508
Change-Id: I332ae3e7cf0483129063c4f520174617ec4ebf1a

ezhulenev committed:
[TF:MLIR] Add support for folding Transpose into FusedBatchNormV3

PiperOrigin-RevId: 296344377
Change-Id: I768b18534b17e8c93994279b12e72650f2f0858c

renjie-liu committed:
Update ophint optional inputs logic (fill in optional input with constant nodes), also update the toco unidirectional lstm legalization.

PiperOrigin-RevId: 296345130
Change-Id: I5b2b8262a9495ff4f3c46f35724c030a914949f6

terryheo committed:
Fix crashing on GPU elementwise ops

Pass ElementwiseAttributes para as a pointer to check if it's valid or not.

PiperOrigin-RevId: 296348071
Change-Id: Ia0a4149605d5fbff5f6a08176ea7eb004bb23315

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 296351967
Change-Id: I84b026ad9fc32992818caa452fede88732faae39

ezhulenev committed:
[TF:MLIR] Add support for folding Transpose into Mean

PiperOrigin-RevId: 296361326
Change-Id: I677bfd6aa17865514a8770b49bce6b7681d5c289

blakehechtman committed:
[XLA] Add some more slice of pad optimizations.

PiperOrigin-RevId: 296361878
Change-Id: I4dbef5e94d95f3337c1004e8c3f09c7a94148075

dayeongl committed:
Fix ProfilingListener for subclasses to override.
Fix BenchmarkTfLiteModel to pass ProfileSummaryFormatter to ProfilingListener.

PiperOrigin-RevId: 296362673
Change-Id: I9e494202c03d8794effdf11eb1bdf1f69d62d35c

ezhulenev committed:
[TF:MLIR] Make Relu layout agnostic operation

PiperOrigin-RevId: 296364496
Change-Id: I86e5f2057984f85333f39f618beeda6cc862afad

gunan committed:
Avoid depending on the implementation of jit:flags in pywrap_tfe.

It was causing the IsXlaEnabled function to return false erroneously.

PiperOrigin-RevId: 296368921
Change-Id: I22507c7fa4bcf8804a333f4eafe38d4c009b76d2

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 296372660
Change-Id: I2725f20a3e95307c13a765565a8f055525827687

Commit:
Stop the gradient for QuantizeAndDequantizeV2 when the input is out of range.

PiperOrigin-RevId: 296372785
Change-Id: Ia0a6168dac58a9a04183a4aa7da93cec231f5fb1

tensorflower-gardener committed:
Stop the gradient for QuantizeAndDequantizeV2 when the input is out of range.

PiperOrigin-RevId: 296377159
Change-Id: Ib92e49c557f85c14f07f67a18814000d71ff5134

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 296385138
Change-Id: I7c823614efbc51b718b4435426040fa369146174

tensorflower-gardener committed:
Merge pull request #36944 from Intel-tensorflow:nhasabni/mkl_eager_op_rewrite_fix_v2

PiperOrigin-RevId: 296386102
Change-Id: I4ef2f4d104011ff01ce050eac7041827d2c907b6

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2020-02-21

PiperOrigin-RevId: 296386877
Change-Id: I121078b4d184fcd042a91f0a10c231e313223b03

VoVAllen committed:
fix android and windows building

VoVAllen committed:
space

buchgr committed:
cleanup: delete unused ubuntu14.04 and centos6 preconfigs

PiperOrigin-RevId: 296393046
Change-Id: I7e18e90860d8a1bd072956efe7dcfd4cab6af1bf

Commit:
Remove left-over debugging.

PiperOrigin-RevId: 296396471
Change-Id: I5478bef24e53a2eb80f4d84774c8bfa58273fc6f

Commit:
Add support to lower element-wise HLO functions to LHLO on dynamic shapes.

This uses hand-written shape derivation functions and is not complete. The goal is to start experimenting with these and get some first code through.

PiperOrigin-RevId: 296403560
Change-Id: I996cc5f862604ca76344076cb6699998757d4164

galarragas committed:
Support partition limitation even when accelerator name is not specified

PiperOrigin-RevId: 296406206
Change-Id: I4bdd9d0e9b578401bb380828c628e14ef5aa711c

VoVAllen committed:
fix sanity check

VoVAllen committed:
Merge branch 'master' into dlpack

VoVAllen committed:
fix leak

VoVAllen committed:
fix
